### DISCUSSION TITLE
Using AIOBLE to connect to multiple BLE devices

### DISCUSSION THREAD

**Discussion by JonLaws** [micropython/micropython#13162] [Using MicroPython]
I'm trying to use AIOBLE to connect to multiple bluetooth peripherals.
The aioble example scripts [https://github.com/micropython/micropython-lib/tree/master/micropython/bluetooth/aioble](url) have an "async with connection" section.
If I try to start a second connection whilst the first connection is in the "async with connection" section of the code, it crashes.
Anybody know what I'm doing wrong or alternatively have some example code for AIOBLE to connect to multiple bluetooth peripherals?

**Comment by brianreinhold** [comment]
My understanding of this problem is that first you have to configure whatever Bluetooth stack you are using to support more than one simultaneous connection. I configured the btstack for the PICO W to support at least three simultaneous connections. This is likely what you are not doing. I got crashes as well prior to configuring the stack for more than one device. Then you have to allow the scanner to restart after connecting, as the scanner is stopped during connecting.
